src/com/owncloud/android/lib/resources/files/ChunkedUploadRemoteFileOperation.java

```diff
@@ -99,6 +100,12 @@ protected int uploadFile(OwnCloudClient client) throws IOException {
                 mPutMethod.addRequestHeader(OC_CHUNKED_HEADER, OC_CHUNKED_HEADER);
                 mPutMethod.addRequestHeader(OC_CHUNK_SIZE_HEADER, chunkSizeStr);
                 mPutMethod.addRequestHeader(OC_TOTAL_LENGTH_HEADER, totalLengthStr);
+
+                // Tell to the server what is the last modification date of the file to upload
+                Long timeStampLong = System.currentTimeMillis()/1000;
```

Comment:
This is not the date we need to set. We want to read the modification time from the file in the local filesystem with ```java.io.File#lastModified()```. With this code the date set is the date of the start of the upload / chunk instead of the date of the end of the upload.